batchflow/models/torch/utils.py

```diff
@@ -118,3 +120,48 @@ def center_crop(inputs, target_shape, dims):
     crops_ = [slice(crop_lefts[i], crop_lefts[i] + target_shape[i]) for i in range(dims)]
     crops = [slice(dim) for dim in no_crop_shape] + crops_
     return inputs[crops]
+
+
+def get_blocks_and_activations(model):
+    """ Retrieve intermediate blocks of the neural network model
+    and corresponding activation names.
+    """
+    encoder_blocks = list(filter(lambda x: 'block' in x, model.model.encoder))
+    decoder_blocks = list(filter(lambda x: 'block' in x, model.model.decoder))
+
+    blocks = [f'model.encoder["{block}"]' for block in encoder_blocks]
+    embedding = 'embedding' in model.config['order']
+    if embedding:
+        blocks += ['model.embedding']
+    blocks += [f'model.decoder["{block}"]' for block in decoder_blocks]
+
+    activation_names = [f'encoder_{i}' for i in range(len(encoder_blocks))]
+    if embedding:
+        activation_names += ['embedding']
+    activation_names += [f'decoder_{i}' for i in range(len(decoder_blocks))]
+
+    return blocks, activation_names
+
+def compress_activations(batch, activation_names, **kwargs):
```

Comment:
`batch` has no relation to `TorchModel` and will not be part of the model-related-utils